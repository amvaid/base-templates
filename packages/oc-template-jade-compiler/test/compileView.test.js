jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000;

const path = require('path');
const fs = require('fs-extra');
const compileView = require('../lib/compileView.js');

const componentPath = path.join(__dirname, '../../../mocks/jade-component');
const publishPath = path.join(componentPath, '_package');
const publishFileName = 'template.js';

const options = {
  componentPackage: {
    oc: {
      files: {
        template: {
          src: 'template.jade',
          type: 'jade'
        }
      }
    }
  },
  componentPath,
  publishPath,
  publishFileName
};

test('Should correctly compile the view', done => {
  compileView(options, (err, compiledViewInfo) => {
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs.readFileSync(path.join(publishPath, publishFileName), 'UTF8')
    ).toMatchSnapshot();
    fs.removeSync(path.join(publishPath, publishFileName));
    done();
  });
});
