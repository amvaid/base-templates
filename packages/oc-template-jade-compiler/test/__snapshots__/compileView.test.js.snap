// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the view 1`] = `
Object {
  "hashKey": "43155d7e8788d3a8b32cc7aeb4c7ff1c30d83ba7",
  "src": "template.js",
  "type": "jade",
}
`;

exports[`Should correctly compile the view 2`] = `"var oc=oc||{};oc.components=oc.components||{},oc.components[\\"43155d7e8788d3a8b32cc7aeb4c7ff1c30d83ba7\\"]=function(c){var o,e=[];return e.push(\\"<h1>\\"+jade.escape(null==(o=\\"Hello jade!\\")?\\"\\":o)+\\"</h1>\\"),e.join(\\"\\")};"`;
